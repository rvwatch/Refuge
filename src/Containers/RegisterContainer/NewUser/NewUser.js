import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as Actions from '../../../Actions/';
import { postRegister } from '../../../ApiCalls/postRegister';
import { object, func } from 'prop-types';
import { withRouter } from 'react-router-dom';

export class NewUser extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      name: ''
    };
  }

  handleInput = event => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  registerUser = async event => {
    event.preventDefault();
    const { email, password, name } = this.state;
    const user = await postRegister(email, password, name);
    this.props.addUser(user);
    this.props.setLoggedIn();
    this.props.history.push('/');
  };

  render() {
    return (
      <form className="NewUser-form" onSubmit={this.registerUser}>
        <h2>Create an Account</h2>
        <div>
          <input
            onChange={this.handleInput}
            type="text"
            placeholder="Username"
            name="name"
            value={this.state.name}
          />
        </div>
        <div>
          <input
            onChange={this.handleInput}
            type="text"
            placeholder="email"
            name="email"
            value={this.state.email}
          />
        </div>
        <div>
          <input
            onChange={this.handleInput}
            type="password"
            placeholder="password"
            name="password"
            value={this.state.password}
          />
        </div>
        <div>
          <input className="submit-btn" type="submit" value="enter" />
        </div>
      </form>
    );
  }
}

export const mapStateToProps = ({ user }) => ({ user });

export const mapDispatchToProps = dispatch => ({
  addUser: user => dispatch(Actions.addUser(user)),
  setLoggedIn: () => dispatch(Actions.setLoggedIn()),
  addErrorMessage: error => dispatch(Actions.addErrorMessage(error))
});

NewUser.propTypes = {
  target: object,
  addUser: func,
  setLoggedIn: func,
  history: object,
  addErrorMessage: func
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(NewUser));
