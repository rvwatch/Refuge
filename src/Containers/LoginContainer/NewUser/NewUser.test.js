import {NewUser} from './NewUser';

describe('NewUser', () => {
  it('should match the snapshot', () => {
    expect(NewUser).toMatchSnapshot();
  });
});
