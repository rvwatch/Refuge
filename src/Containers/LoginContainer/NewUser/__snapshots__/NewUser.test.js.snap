// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUser should match the snapshot 1`] = `undefined`;
